### ISSUE TITLE
Option to provide pairs of restart files at specified unsteady write file iterations

### ISSUE THREAD

**Issue by demanosalvas** [su2code/SU2#133]
Currently when you are running an unsteady case and the solution file is not written on every iteration there is no smooth way to restart the simulation. Every restart has to deal with a transient being generated and to fix this it will be good to output pairs of solution files every time it is required to write the solution.

**Comment by omdemircan** [comment]
Hi, I have been running unsteady simulations lately and encountered the issues you mentioned. For the restart, it is possible to duplicate one restart file and use them both as pairs. As for the solution at each iteration issue, SU2_SOL doesn't seem to work properly for unsteady cases. However, you can use the the script below in order to get the Paraview or Tecplot output for each iteration using your config file.
```
#!/usr/bin/python
import fileinput
import sys
import os

ni = 100
nf = 10000
dn = 100

for line in fileinput.input(inplace=True):
	if line.startswith('UNSTEADY_SIMULATION'):
		print('UNSTEADY_SIMULATION= NO')
	else:
		sys.stdout.write(line)

for n in range(ni, nf+1, dn):
	for line in fileinput.input(inplace=True):
		if line.startswith('SOLUTION_FLOW_FILENAME'):
			print('SOLUTION_FLOW_FILENAME= restart_flow_%05d.dat' % n)
		elif line.startswith('SURFACE_FLOW_FILENAME'):
			print('SURFACE_FLOW_FILENAME= surface_flow_%05d' % n)
		else:
			sys.stdout.write(line)
	os.system('SU2_SOL config_SOL.cfg')
```

I use it by running the following on my terminal: python script.py config.cfg
Note that ni is the starting iteration, nf is the last iteration and dn is the interval. You can place this script where your config file is and run it as described. You may want to use a duplicate of your config file since the script will modify it for each iteration.

**Comment by talbring** [comment]
Hi @omdemircan,

I don't have any problems with SU2_SOL generating the visualization files for unsteady cases. Can you be more precise what is not working for you ?

**Comment by omdemircan** [comment]
On my personal computer, I couldn't manage to use SU2_SOL for unsteady cases in order to obtain Paraview output for all iterations. On the other hand, I don't know how to specify an interval for the Paraview output, as mentioned by @demanosalvas . With the script that I shared, I managed to get desired files by introducing the case as steady and changing the file names for each iteration.

**Comment by talbring** [comment]
What version are you using ? Because at least with the latest development version for me also the Paraview output for unsteady problems works fine.

**Comment by talbring** [comment]
You can use the option WRT_SOL_FREQ_DUALTIME to specify an interval for the output.

**Comment by omdemircan** [comment]
I am using SU2 V5. I have been using SU2 for less than 6 months now and the SU2_SOL command doesn't work in parallel for me, it gives a segmentation fault error. Otherwise, I managed to run it in serial as of yesterday but I have to change restart files names to solution file names and make sure the config file restart input matches the first iteration file name.
Secondly, by interval I meant solutions between one iteration and another. For instance, when I want Paraview output from between iterations 10000 and 20000 with steps of 500 iterations, how do I proceed? I think WRT_SOL_FREQ_DUALTIME only lets me define 500 as the step size. How do I input the starting and ending final iteration for postprocessing?

**Comment by MehdiMortazavi1992** [comment]
Dear @talbring and @omdemircan ,

What is the quickest/ easiest way to specify a WRT_SOL_FREQ_DUALTIME for "surface_flow_xxxx.csv" files which would be different from the frequency for "flow_xxxx.plt or vtk etc" the reason is that I would like to have a more frequent number of "surface_flow.csv" files (as they are smaller in size) than the "flow.plt" files (which are much larger and will slow down the simulation time if I were to generate them as frequently as the "surface_flow.csv" files).
Thank you so much!

**Comment by 2456439929** [comment]
Hi, I am trying to use the unsteady simulation part of su2, but I can't find any tutoial about it,  can any one provide me a complete procedure of unsteady simulation (better including mesh deformation part). Thanks a lot!!! @talbring
